src/os/layer/vectortile.js

```diff
@@ -0,0 +1,727 @@
+  }
+
+  /**
+   * @inheritDoc
+   */
+  setRemovable(value) {
+    this.removable_ = value;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  getNodeUI() {
+    return this.nodeUI_;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  setNodeUI(value) {
+    this.nodeUI_ = value;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  getLayerUI() {
+    return this.layerUi_;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  setLayerUI(value) {
+    this.layerUi_ = value;
+  }
+
+  /**
+   * Identify the layer on the map.
+   *
+   * @protected
+   */
+  identify() {
+    identifyLayer(this);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  getLayerVisible() {
+    return this.getVisible();
+  }
+
+  /**
+   * @inheritDoc
+   */
+  setLayerVisible(value) {
+    if (value !== this.getLayerVisible()) {
+      this.setVisible(value);
+      this.dispatchEvent(new PropertyChangeEvent(LayerPropertyChange.VISIBLE, value, !value));
+    }
+  }
+
+  /**
+   * @inheritDoc
+   */
+  setBaseVisible(value) {
+    this.setVisible(value);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  getBaseVisible() {
+    return this.getVisible();
+  }
+
+  /**
+   * @inheritDoc
+   */
+  getLayerOptions() {
+    return this.layerOptions_;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  setLayerOptions(value) {
+    this.layerOptions_ = value;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  callAction(type) {
+    var source = this.getSource();
+    if (source) {
+      switch (type) {
+        case ActionEventType.IDENTIFY:
+          this.identify();
+          break;
+        case ActionEventType.MOST_RECENT:
+          DataManager.getInstance().setTimeFromDescriptor(this.getId());
+          break;
+        case ActionEventType.REFRESH:
+          source.refresh();
+          break;
+        case ActionEventType.REMOVE_LAYER:
+          var removeEvent = new LayerEvent(LayerEventType.REMOVE, this.getId());
+          dispatcher.dispatchEvent(removeEvent);
+          break;
+        case ActionEventType.RENAME:
+          renamelayer.launchRenameDialog(this);
+          break;
+        default:
+          break;
+      }
+    }
+  }
+
+  /**
+   * @inheritDoc
+   */
+  getGroupUI() {
+    return null;
+  }
+
+  /**
+   * @inheritDoc
+   * @see {ui.action.IActionTarget}
+   */
+  supportsAction(type, opt_actionArgs) {
+    switch (type) {
+      case ActionEventType.GOTO:
+        var projExtent = osMap.PROJECTION.getExtent();
+        var layerExtent = reduceExtentFromLayers(/** @type {!ol.Extent} */ (olExtent.createEmpty()), this);
+        var projArea = olExtent.getArea(projExtent);
+        var layerArea = olExtent.getArea(layerExtent);
+        return !olExtent.isEmpty(layerExtent) && layerArea / projArea < 0.8;
+      case ActionEventType.IDENTIFY:
+      case ActionEventType.REFRESH:
+      case ActionEventType.SHOW_DESCRIPTION:
+        return true;
+      case ActionEventType.RENAME:
+        return Array.isArray(opt_actionArgs) && opt_actionArgs.length === 1;
+      case ActionEventType.MOST_RECENT:
```

Comment:
Do VectorTiles support time? Are any of these other actions inapplicable?

Reply:
No they don't, at least not in the request. I removed that action and the others are all supported.